server/server/models/user.py

```diff
@@ -69,3 +72,113 @@ def is_active(self, request):
                                 .filter_by(status="Active").one().id
                 return True
         return False
+
+    def get_role(self):
+        """Return string with user role"""
+        return self.roles.role
+
+    def __init__(self, email, nickname, password, create_date, location,
+                 first_name, last_name, status_id, role_id, avatar,
+                 banned_to_date):
+        self.email = email
+        self.nickname = nickname
+        self.password = password
+        self.create_date = create_date
+        self.location = location
+        self.first_name = first_name
+        self.last_name = last_name
+        self.status_id = status_id
+        self.role_id = role_id
+        self.avatar = avatar
+        self.banned_to_date = banned_to_date
+
+
+def from_json(cls, data):
+    """ Deserializes User object """
+    return cls(**data)
+
+
+def model_to_json(sqlalchemy_object):
+    """ Serializes sqlalchemy_object to json
+    and converts datetime to string
+    """
+    fields_arr = [prop.key for prop in
+                  class_mapper(sqlalchemy_object.__class__).iterate_properties
+                  if isinstance(prop, ColumnProperty)]
+    _dict = {}
+    for key in fields_arr:
+        temp = getattr(sqlalchemy_object, key)
+        if isinstance(temp, datetime.datetime):
+            _dict[key] = str(getattr(sqlalchemy_object, key))
+        else:
+            _dict[key] = getattr(sqlalchemy_object, key)
+    return _dict
+
+
+def users_to_json(obj):
+    """ Specific serializer for User object (json) """
+    to_serialize = ['id', 'email', 'nickname', 'password', 'create_date',
+                    'location', 'first_name', 'last_name', 'status_id',
+                    'role_id', 'avatar', 'banned_to_date']
+    dict_ = {}
+    for attr_name in to_serialize:
+        d[attr_name] = str(getattr(obj, attr_name))
+    return dict_
+
+
+class DTEncoder(json.JSONEncoder):
+    """ Decoder class for datetime (json) """
+    def default(self, obj):
+        if isinstance(obj, datetime.datetime):
+            return o.isoformat()
+        return json.JSONEncoder.default(self, obj)
+
+
+def get_all_users(request):
+    """ Method to get all users from database """
+    qry = request.dbsession.query(User).all()
+    usr_in_json = list(map(lambda x: model_to_json(x), qry))
+    return usr_in_json
+
+
+def create_usr(json_str, request):
+    """ Method to add user to database """
+    user = User(**json_str)
+    request.dbsession.add(user)
+    return {'status': 'Success!'}
+
+
+def read_usr(request):
+    """ Method to read user from database """
+    if request.dbsession.query(User)\
+            .filter(User.id == request.matchdict['profile_id']).count():
+        qry = request.dbsession.query(User)\
+            .filter(User.id == request.matchdict['profile_id']).first()
```

Comment:
I'd better use [get](http://docs.sqlalchemy.org/en/latest/orm/query.html#sqlalchemy.orm.query.Query.get) method to take a model by id. And you don't need to check presence before fetching model. Just `get` it, and if it doesn't exist you will receive None.